src/core/Akka.Persistence.Tests/PersistentActorRecoveryTimeoutSpec.cs

```diff
@@ -0,0 +1,173 @@
+using Xunit;
+
+namespace Akka.Persistence.Tests
+{
+
+    public class PersistentActorRecoveryTimeoutSpec : PersistenceSpec
+    {
+        private const string JournalId = "persistent-actor-recovery-timeout-spec";
+
+        #region internal test classes
+
+        internal class RecoveryTimeoutActor : NamedPersistentActor
+        {
+            private readonly IActorRef _probe;
+
+            public RecoveryTimeoutActor(IActorRef probe) : base("recovery-timeout-actor")
+            {
+                _probe = probe;
+            }
+
+            protected override bool ReceiveRecover(object message)
+            {
+                return true;
+            }
+
+            protected override bool ReceiveCommand(object message)
+            {
+                Persist(message, x => Sender.Tell(message));
+                return true;
+            }
+
+            protected override void OnRecoveryFailure(Exception reason, object message = null)
+            {
+                _probe.Tell(new Status.Failure(reason));
+            }
+        }
+
+        internal class ReceiveTimeoutActor : NamedPersistentActor
+        {
+            private readonly TimeSpan _receiveTimeout;
+            private readonly IActorRef _probe;
+
+            public ReceiveTimeoutActor(TimeSpan receiveTimeout, IActorRef probe) : base("recovery-timeout-actor-2")
+            {
+                _receiveTimeout = receiveTimeout;
+                _probe = probe;
+            }
+
+            protected override void PreStart()
+            {
+                base.PreStart();
+                Context.SetReceiveTimeout(_receiveTimeout);
+            }
+
+            protected override bool ReceiveRecover(object message)
+            {
+                if (message is RecoveryCompleted)
+                    _probe.Tell(Context.ReceiveTimeout);
+                return true;
+            }
+
+            protected override bool ReceiveCommand(object message)
+            {
+                Persist(message, x => Sender.Tell(message));
+                return true;
+            }
+
+            protected override void OnRecoveryFailure(Exception reason, object message = null)
+            {
+                _probe.Tell(new Status.Failure(reason));
+            }
+        }
+
+        #endregion
+
+        public PersistentActorRecoveryTimeoutSpec()
+            : base(SteppingMemoryJournal.Config(JournalId).WithFallback(
+                ConfigurationFactory.ParseString(
+                    @"akka.persistence.journal.stepping-inmem.recovery-event-timeout = 1000ms
+                    akka.actor.serialize-messages = off"))
+                  .WithFallback(Configuration("PersistentActoryRecoveryTimeoutSpec")))
+        {
+            Sys.EventStream.Publish(new Mute(new CustomEventFilter(e => e.Message is AskTimeoutException)));
+            //TODO: remove akka.actor.serialize-messages=off when Props serialization will be resolved (github issue: #569)
```

Comment:
This issue has been closed a long time ago. 
https://github.com/akkadotnet/akka.net/issues/569